### PR TITLE
fix: Remove embedded node dependency

### LINKED ISSUE #486: Remove embedded node and parachain dependencies
**Is your feature request related to a problem? Please describe.**
There are dependencies on our parachain crates including the runtime which can cause upgrade issues since subxt can have conflicting sub-dependencies.

**Describe the solution you'd like**
Remove the embedded node (`runtime/client`) and instead use a local node with `instant-seal`. We already do this for `bitcoind` and `electrs` (see `docker-compose.yml`). This is also the testing strategy adopted by `subxt` so there should be examples upstream for how to do this. Remove any other parachain dependencies and re-declare types if necessary. The only necessary import is probably the Bitcoin parsing for which we can minimize the dependencies upstream.

**Describe alternatives you've considered**
The current alternative is to run an embedded node which is adds substantial compilation time since we also need to build the parachain runtime code.

### LINKED ISSUE #497: Remove embedded node dependency
**Is your feature request related to a problem? Please describe.**
There are dependencies on our parachain crates including the runtime which can cause upgrade issues since subxt can have conflicting sub-dependencies.

**Describe the solution you'd like**
Remove the embedded node (runtime/client) and instead use a local node with an instant seal.

**Describe alternatives you've considered**
The current alternative is to run an embedded node, which adds substantial compilation time since we also need to build the parachain runtime code.

### PR DESCRIPTION
**Pull request by nakul1010**
Closes #497
The PR adds the following things
- Starts and kills the para chain docker service for required test cases
- Runs the Parachain dependent test cases serially to avoid clashes

### CODE REVIEW

File: runtime/src/integration/mod.rs [hunk 1]
```diff
@@ -223,6 +191,13 @@ pub async fn with_timeout<T: Future>(future: T, duration: Duration) -> T::Output
     timeout(duration, future).await.expect("timeout")
 }
 
+pub async fn start_chain() -> std::io::Result<Child> {
+    let command = Command::new("sh").arg("../scripts/run_parachain_node.sh").spawn();
```
Comment (gregdhill) [inline runtime/src/integration/mod.rs:195]: Is it possible to keep the parachain running and refactor the tests to support using existing state?

Comment (gregdhill) [reply runtime/src/integration/mod.rs:195]: Even with `instant-seal` it takes some time to startup which we could avoid.

File: runtime/src/integration/mod.rs [hunk 2]
```diff
@@ -223,6 +191,13 @@ pub async fn with_timeout<T: Future>(future: T, duration: Duration) -> T::Output
...
+    // Time to start the parachain image, so RPC request can be handle
+    tokio::time::sleep(Duration::from_secs(2)).await;
```
Comment (gregdhill) [inline runtime/src/integration/mod.rs:197]: I don't like sleepers such as this, we already had a big problem in the Typescript library with these. Where possible I would prefer to await some expected state such the result of making a successful RPC connection.

Comment (gregdhill) [reply runtime/src/integration/mod.rs:197]: I think `from_url_with_retry` should already do that, if there is a specific test case that fails maybe reorder the setup logic.

File: vault/tests/vault_integration_tests.rs [hunk 3]
```diff
@@ -71,21 +71,23 @@ where
...
 
-    execute(client, vault_id, vault_provider).await
+    execute(parachain_rpc, vault_id, vault_provider).await
 }
 
 #[tokio::test(flavor = "multi_thread")]
+#[serial]
 async fn test_redeem_succeeds() {
+    let mut parachain_runner: Child = start_chain().await.unwrap();
     test_with_vault(|client, vault_id, vault_provider| async move {
```
Comment (gregdhill) [inline vault/tests/vault_integration_tests.rs:88]: Could we move `start_chain()` into `test_with_vault`?

File: runtime/src/integration/mod.rs (hunk 1 above)
Comment (nakul1010) [reply runtime/src/integration/mod.rs:195]: I have researched it, and the solution is 
- To set up Chain Storage to the `Genesis state` for every test case. However, this approach has its own limitations as any changes in runtime business logic need to update the `set_genesis_state` method on the client side accordingly.

File: runtime/src/integration/mod.rs (hunk 2 above)
Comment (nakul1010) [reply runtime/src/integration/mod.rs]: Removed the sleep duration and stopped the old parachain instance before running the script in the background thread.

**Review by sander2 — APPROVED** [review]

**Pull request merged**